src/status_im/common/biometric/events.cljs

```diff
@@ -1,120 +1,124 @@
-
-(defn get-label-by-type
-  [biometric-type]
-  (condp = biometric-type
-    constants/biometrics-type-android (i18n/label :t/biometric-fingerprint)
-    constants/biometrics-type-face-id (i18n/label :t/biometric-faceid)
-    (i18n/label :t/biometric-touchid)))
-
-(defn get-icon-by-type
-  [biometric-type]
-  (condp = biometric-type
-    constants/biometrics-type-face-id :i/face-id
-    :i/touch-id))
-
-(re-frame/reg-fx
- :biometric/get-supported-biometric-type
+(rf/reg-fx
+ :biometric/get-supported-type
  (fn []
    ;;NOTE: if we can't save user password, we can't use biometric
    (keychain/can-save-user-password?
     (fn [can-save?]
-      (when (and can-save? (not android-device-blacklisted?))
+      (when (and can-save? (not utils/android-device-blacklisted?))
         (-> (biometrics/get-supported-type)
             (.then (fn [type]
-                     (rf/dispatch [:biometric/get-supported-biometric-type-success type])))))))))
+                     (rf/dispatch [:biometric/set-supported-type type])))))))))
 
-(rf/defn get-supported-biometric-auth-success
-  {:events [:biometric/get-supported-biometric-type-success]}
-  [{:keys [db]} supported-type]
+(defn set-supported-type
+  [{:keys [db]} [supported-type]]
   {:db (assoc db :biometric/supported-type supported-type)})
 
-(rf/defn show-message
-  {:events [:biometric/show-message]}
-  [_ error]
-  (let [code    (ex-cause error)
-        content (if (#{::biometrics/not-enrolled
+(schema/=> set-supported-type events-schema/?set-supported-type)
+(rf/reg-event-fx :biometric/set-supported-type set-supported-type)
+
+(defn show-message
+  [_ [code]]
+  (let [content (if (#{::biometrics/not-enrolled
                        ::biometrics/not-available}
                      code)
                   (i18n/label :t/grant-face-id-permissions)
                   (i18n/label :t/biometric-auth-error {:code code}))]
-    {:effects.utils/show-popup
-     {:title   (i18n/label :t/biometric-auth-login-error-title)
-      :content content}}))
+    {:fx [[:effects.utils/show-popup
+           {:title   (i18n/label :t/biometric-auth-login-error-title)
+            :content content}]]}))
 
+(schema/=> show-message events-schema/?show-message)
+(rf/reg-event-fx :biometric/show-message show-message)
 
-(re-frame/reg-fx
+(rf/reg-fx
  :biometric/authenticate
- (fn [{:keys [on-success on-fail prompt-message]}]
+ (fn [{:keys [on-success on-fail on-cancel prompt-message on-done]}]
    (-> (biometrics/authenticate
         {:prompt-message          (or prompt-message (i18n/label :t/biometric-auth-reason-login))
          :fallback-prompt-message (i18n/label
                                    :t/biometric-auth-login-ios-fallback-label)
          :cancel-button-text      (i18n/label :t/cancel)})
        (.then (fn [not-canceled?]
-                (when (and on-success not-canceled?)
-                  (on-success))))
+                (utils/handle-cb on-done)
```

Comment:
This function name handle-cb really lacks any description, can the function name state a bit more of what it does

Reply:
removed it entirely